backend/cmd/headlamp.go

```diff
@@ -719,11 +725,72 @@ func handleClusterHelm(c *HeadlampConfig, router *mux.Router) {
 	})
 }
 
+// handleWebsocket handles websocket requests. It returns context key
+// which is used to store the context in the cache. The context key is
+// unique for each user. It is found in the "X-HEADLAMP-USER-ID" parameter
+// in the websocket URL.
+func handleWebsocket(w http.ResponseWriter, r *http.Request, clusterName string) string {
+	var contextKey string
+	// Parse the URL
+	u, err := url.Parse(r.URL.String())
+	if err != nil {
+		log.Println("Error: parsing URL: ", err)
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+
+		return ""
+	}
+
+	// Get the query parameters
+	queryParams := u.Query()
+	// Check if "X-HEADLAMP-USER-ID" parameter is present in the websocket URL.
+	userIDparam := queryParams.Get("X-HEADLAMP-USER-ID")
```

Comment:
It was discussed internally between @yolossn and myself. When there is a websocket connection we cannot send kubeconfig as header like we are doing in a normal request. Websocket do not support it. Websockets connection are only open when a cluster proxy has been saved in cache and when a user tries to open let say home page of a particular cluster. Since we didn't have kubeconfig sent from the cluster we needed a way to know whether the cluster that user has clicked on was a normal cluster or stateless one. 

This param isolate's stateless cluster created by one user alone.

Reply:
Ok, I will send it as a message to add security. Thanks for the suggestion

Reply:
@illume , I want to understand this issue better.
1. Can the info be leaked only in non-secure connections?
2. Can someone get straight access to the cluster, or just be able to see that the cluster exists (but they still require a token)?

We are hoping no-one exposes Headlamp to the internet through a non-secure connection of course.

Reply:
This is now fixed. We are now not using query parameters anymore. We now send the uuid using websocket protocol headers which solves all the security issue. Please refer https://github.com/headlamp-k8s/headlamp/pull/1303/commits/36613c84b6e7830c8061f8fcef715bb81590f592